app/services/irs_attempts_api/event.rb

```diff
@@ -0,0 +1,101 @@
+module IrsAttemptsApi
+  class Event
+    attr_reader :jti, :iat, :event_type, :encrypted_event_data
+
+    def initialize(jti:, iat:, event_type:, encrypted_event_data:)
+      @jti = jti
+      @iat = iat
+      @event_type = event_type
+      @encrypted_event_data = encrypted_event_data
+    end
+
+    def self.build(event_type:, session_id:, occurred_at:, event_metadata:, now: Time.zone.now)
+      jti = SecureRandom.uuid
+      iat = now.to_i
+      encrypted_event_data = encrypt_event_data(
+        session_id: session_id,
+        occurred_at: occurred_at,
+        event_metadata: event_metadata,
+      )
+      self.new(
+        jti: jti, iat: iat, event_type: event_type, encrypted_event_data: encrypted_event_data,
+      )
+    end
+
+    def security_event_token_data
+      @security_event_token_data ||= {
+        'iss' => Rails.application.routes.url_helpers.root_url,
+        'jti' => jti,
+        'iat' => iat,
+        'aud' => IdentityConfig.store.irs_attempt_api_audience,
+        'events' => events,
+      }
+    end
+
+    def jwt
+      JWT.encode(
+        security_event_token_data,
+        AppArtifacts.store.oidc_private_key,
```

Comment:
We did add add some info to our certs presenter that says we only use that key for signing... so that should be fine? I assume if we went the JWE route, we would encrypt to their public key (instead of encrypt with our private key again)
https://github.com/18f/identity-idp/pull/6007

Reply:
Yeah, I just pushed a commit to jump over to JWE and all of the signing logic is gone now in favor of encrypting with their private key